### DISCUSSION TITLE
Type error when using with react and typescript

### DISCUSSION THREAD

**Discussion by lilkahuna** [emailjs-com/help#29] [General]
Has anyone used email js with react and typescript, I keep getting an error with the useRef hook which says I need it to be set to HTMLFormElement. I gave the useRef hook the type it siad it wanted but still got an error. Does anyone have a solution or fix for this?

Error:
The 3rd parameter is expected to be the HTML form element or the style selector of form

**Comment by xr0master** [comment]
```ts
const formRef = useRef<HTMLFormElement>(null);
```

**Reply by lilkahuna** [reply]
I still get an error of argument of type null is not assignable to parameter of type string | HTMLFormElement. Type null is not assignable to type string | HTMLFormElement

**Reply by xr0master** [reply]
There are many ways to solve your syntax issue, for example as below. 
```ts
formRef.current!
```

Keep in mind that you are better off asking questions about TypeScript in the TS community.
